### ISSUE TITLE
Averagepool : using "np.nan" in padding instead of "zeros"

### ISSUE THREAD

**Issue by amitkinhikar** [onnx/onnx#1384]
In averagepool example "averagepool_2d_pads" while padding, it is using "np.nan" which is giving average as very high number, but in another example "averagepool_2d_pads_count_include_pad" "constant value = 0" is used. I think the np.nan should be replaced with zero. 

Could you please clarify on this.

**Comment by amitkinhikar** [comment]
Also, setting/unsetting of "count_include_pad" does not change in output. It seems that the padded elements are counted for averaging by default.

**Comment by fumihwh** [comment]
@amitkinhikar 
For `count_include_pad=0` (default value), I use `np.nan` here for locating pads exactly by using `np.isnan`. Because input could contain zero value commonly.
For `count_include_pad=1`, use zero pad is okay because we will count them anyway.
By the way, the test case is just for testing. In normal usage, pad means pad with 0.

**Comment by fumihwh** [comment]
@amitkinhikar 
I have to say, two calculations are equivalent.
And I come from tensorflow so I referred to https://www.tensorflow.org/api_guides/python/nn#Pooling
In pytorch or caffe2, they use `floor` instead of `ceil` in default. They have `ceil mode` also.

**Comment by amitkinhikar** [comment]
Thanks @fumihwh for clarification !

**Comment by houseroad** [comment]
Thanks @fumihwh
